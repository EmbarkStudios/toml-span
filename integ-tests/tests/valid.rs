use integ_tests::valid;

valid!(comments);
valid!(evil);

mod arrays {
    use super::valid;

    valid!(empty, "thevoid = [[[[[]]]]]");
    valid!(no_spaces, "ints = [1,2,3]");
    valid!(heterogenous, r#"mixed = [[1, 2], ["a", "b"], [1.1, 2.1]]"#);
    valid!(nested, r#"nest = [["a"], ["b"]]"#);
    valid!(ints_and_floats, "ints-and-floats = [1, 1.1]");
    valid!(
        ints_and_arrays,
        r#"arrays-and-ints =  [1, ["Arrays are not integers."]]"#
    );
    valid!(strings_and_ints, r#"strings-and-ints = ["hi", 42]"#);
    valid!(
        one,
        "[[people]]\nfirst_name = \"Bruce\"\nlast_name = \"Springsteen\""
    );
}

mod tables {
    use super::valid;

    valid!(
        implicit_and_explicit_after,
        r"
[a.b.c]
answer = 42

[a]
better = 43
"
    );
    valid!(
        implicit_and_explicit_before,
        r"
[a]
better = 43

[a.b.c]
answer = 42
"
    );
    valid!(implicit_groups, "[a.b.c]\nanswer = 42");
    valid!(implicit_array, "[[albums.songs]]\nname = \"Glory Days\"");
    valid!(
        array_many,
        r#"
[[people]]
first_name = "Bruce"
last_name = "Springsteen"

[[people]]
first_name = "Eric"
last_name = "Clapton"

[[people]]
first_name = "Bob"
last_name = "Seger"
"#
    );
    valid!(
        nested_arrays,
        r#"
[[albums]]
name = "Born to Run"

  [[albums.songs]]
  name = "Jungleland"

  [[albums.songs]]
  name = "Meeting Across the River"

[[albums]]
name = "Born in the USA"
  
  [[albums.songs]]
  name = "Glory Days"

  [[albums.songs]]
  name = "Dancing in the Dark"
"#
    );
    valid!(sub_empty, "[a]\n[a.b]");
    valid!(table_span, "\n[workspace]\nmembers = true\nother = false\n[workspace.foo]\nbar=3\n\n[workspace.bar]\ninline = { foo = 3 }\n\n");
}

mod numbers {
    use super::valid;

    valid!(integers);
    valid!(floats);
}
