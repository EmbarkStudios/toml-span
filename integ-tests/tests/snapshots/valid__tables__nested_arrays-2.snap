---
source: integ-tests/tests/valid.rs
expression: spans
---
note[string]: root_albums_0_name
  ┌─ nested_arrays:3:9
  │
3 │ name = "Born to Run"
  │         ^^^^^^^^^^^

note[string]: root_albums_0_songs_0_name
  ┌─ nested_arrays:6:11
  │
6 │   name = "Jungleland"
  │           ^^^^^^^^^^

note[table]: root_albums_0_songs_0
  ┌─ nested_arrays:5:3
  │  
5 │ ╭   [[albums.songs]]
6 │ │   name = "Jungleland"
  │ ╰─────────────────────^

note[string]: root_albums_0_songs_1_name
  ┌─ nested_arrays:9:11
  │
9 │   name = "Meeting Across the River"
  │           ^^^^^^^^^^^^^^^^^^^^^^^^

note[table]: root_albums_0_songs_1
  ┌─ nested_arrays:8:3
  │  
8 │ ╭   [[albums.songs]]
9 │ │   name = "Meeting Across the River"
  │ ╰───────────────────────────────────^

note[array]: root_albums_0_songs
  ┌─ nested_arrays:5:3
  │  
5 │ ╭   [[albums.songs]]
6 │ │   name = "Jungleland"
  │ ╰─────────────────────^

note[table]: root_albums_0
  ┌─ nested_arrays:2:1
  │  
2 │ ╭ [[albums]]
3 │ │ name = "Born to Run"
  │ ╰────────────────────^

note[string]: root_albums_1_name
   ┌─ nested_arrays:12:9
   │
12 │ name = "Born in the USA"
   │         ^^^^^^^^^^^^^^^

note[string]: root_albums_1_songs_0_name
   ┌─ nested_arrays:15:11
   │
15 │   name = "Glory Days"
   │           ^^^^^^^^^^

note[table]: root_albums_1_songs_0
   ┌─ nested_arrays:14:3
   │  
14 │ ╭   [[albums.songs]]
15 │ │   name = "Glory Days"
   │ ╰─────────────────────^

note[string]: root_albums_1_songs_1_name
   ┌─ nested_arrays:18:11
   │
18 │   name = "Dancing in the Dark"
   │           ^^^^^^^^^^^^^^^^^^^

note[table]: root_albums_1_songs_1
   ┌─ nested_arrays:17:3
   │  
17 │ ╭   [[albums.songs]]
18 │ │   name = "Dancing in the Dark"
   │ ╰──────────────────────────────^

note[array]: root_albums_1_songs
   ┌─ nested_arrays:14:3
   │  
14 │ ╭   [[albums.songs]]
15 │ │   name = "Glory Days"
   │ ╰─────────────────────^

note[table]: root_albums_1
   ┌─ nested_arrays:11:1
   │  
11 │ ╭ [[albums]]
12 │ │ name = "Born in the USA"
   │ ╰────────────────────────^

note[array]: root_albums
  ┌─ nested_arrays:2:1
  │  
2 │ ╭ [[albums]]
3 │ │ name = "Born to Run"
  │ ╰────────────────────^

note[table]: root
   ┌─ nested_arrays:1:1
   │  
 1 │ ╭ 
 2 │ │ [[albums]]
 3 │ │ name = "Born to Run"
 4 │ │ 
   · │
18 │ │   name = "Dancing in the Dark"
19 │ │ 
   │ ╰^
